collectors/feed/collectors.py

```diff
@@ -0,0 +1,281 @@
+                    else:
+                        text = html_to_text(entry.summary)
+            except Exception:
+                log.exception(
+                    "Failed (processing error) to get feed entry text for: %s",
+                    entry.link,
+                )
+                skip_processing = True
+
+            if not text:
+                # TODO: Consider retrieving text using modified get_full_feed_text for node.js
+                log.info("No text for feed entry: %s", entry.link)
+                skip_processing = True
+
+            if skip_processing:
+                # Prevent reprocessing next time
+                log.info("Skipping feed entry: %s", entry.link)
+                feed_models.append(
+                    SkippedFeedEntry(
+                        url=entry.link,
+                        skipped=True,
+                    )
+                )
+                continue
+
+            cves = find_cves(entry.title + text)
+            cwes = find_cwes(entry.title + text)
+            cvss = find_cvss(text)
+
+            blocklist, allowlist = check_keywords(entry.title + " " + text)
+            allowed = True
+            if blocklist and not allowlist:
+                log.info(
+                    "Skipping %s feed entry (%s) because of blocklist keywords (%s).",
+                    self.feed_name,
+                    entry.title,
+                    "; ".join(blocklist),
+                )
+                allowed = False
+
+            log.info(
+                "Continuing with %s feed entry (%s).",
+                self.feed_name,
+                entry.title,
+            )
+            feed_models.append(
+                FeedEntry(
+                    url=entry.link,
+                    cves=cves,
+                    title=entry.title,
+                    text=text.strip(),
+                    name=self.feed_name,
+                    source=self.source,
+                    components=self.components,
+                    cwe=cwes,
+                    cvss3=cvss,
+                    allowed=allowed,
+                    skipped=False,
+                )
+            )
+
+        log.info(
+            "Found %s new %s feed entries (unfiltered).",
+            len(feed_models),
+            self.feed_name,
+        )
+        count_allowed = len([f for f in feed_models if not f.skipped and f.allowed])
+        log.info(
+            "Found %s new %s feed entries (after filtering).",
+            count_allowed,
+            self.feed_name,
+        )
+
+        return feed_models
+
+
+def run_scraper(feed_data):
+    """Scrape a particular feed."""
+    feed_name = feed_data.get("name")
+    url = feed_data.get("url")
+    source = feed_data.get("source")
+
+    if not url or not feed_name or not source:
+        raise ValueError(
+            "Missing mandatory argument (url, feed name, source) in feed data: %s",
+            feed_data,
+        )
```

Comment:
Considered and ended up not using it. Including a schema would make the MR substantially larger. Instead of explicit error checking, used the square bracket operator / `__getitem__` so that an exception would be raised on some missing items. Since the config is stored in the exact same repo as the code, it doesn't make sense to have a double standard and protect from unskilled/uninformed/asinine changes in one file and not in the rest.